clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -18479,6 +18479,28 @@ Value *CodeGenFunction::EmitAMDGPUBuiltinExpr(unsigned BuiltinID,
         CGM.getIntrinsic(Intrinsic::amdgcn_update_dpp, Args[0]->getType());
     return Builder.CreateCall(F, Args);
   }
+  case AMDGPU::BI__builtin_amdgcn_permlane16:
+  case AMDGPU::BI__builtin_amdgcn_permlanex16: {
+    llvm::Value *Src0 = EmitScalarExpr(E->getArg(0));
```

Comment:
If there's really not a helper to just EmitScalarExpr for N arguments, there should be one used here

Reply:
added a new helper